### PR TITLE
Hamburger Menu Implementation

### PR DESCRIPTION
**Pull request by internette**
Created hamburger menu and uses the data from the homepage to fill the navigation.

image
image
image

### CODE REVIEW

**Comment by jjm3x3** [comment]
Friendly suggestion on "process" can you make sure this links to the corresponding issue #86 that way when this gets merged it will update the status of that issue, also the pages can blinked together. Also I would say this PR is a good use-case for the "test: UAT" label.

Quick UAT related thing, I see that when one is viewing it from a smaller view port the drop-down can cover up the main content like in this screenshot
image

File: app/components/navigation/navigation-item.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { useState } from "react";
+import INavigationItem from '@/app/models/INavigationItem';
+export default function NavigationItem({ inputAttr, labelAttr, listAttr, children, hasSubpages }: INavigationItem) {
+    const [isHidden, setIsHidden] = useState(true);
+    return (<>
+        <input id={inputAttr.id} className={inputAttr.classes} type="checkbox" onChange={(e)=> setIsHidden(!e.target.value)}/>
```
Comment (jjm3x3) [inline app/components/navigation/navigation-item.tsx:6]: nit: I would say formatting this so that it can be more friendly to read on narrow screens can be a nice optimization in code quality and review. It can also make diffs easier to read.
```suggestion
        <input
            id={inputAttr.id}
            className={inputAttr.classes}
            type="checkbox"
            onChange={(e)=> setIsHidden(!e.target.value)}
        />
```

File: app/components/navigation/navigation-item.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+        <label htmlFor={inputAttr.id} id={labelAttr.id} aria-controls={labelAttr.aria.controls} data-testid={labelAttr.testId} className={labelAttr.classes }>{labelAttr.content}</label>
```
Comment (jjm3x3) [inline app/components/navigation/navigation-item.tsx:7]: nit: same as above, though, I am less sure how to fix this one since the tag has content 😅

File: app/components/navigation/navigation-item.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+        {hasSubpages && <ul className={listAttr.classes}  id={listAttr.id} data-testid={listAttr.testId} aria-hidden={isHidden} hidden={isHidden}>{children}</ul>}
+    </>)
+}
```
Comment (jjm3x3) [inline app/components/navigation/navigation-item.tsx:10]: nit: do we not expect newlines EOF yet 😅

**Review by jjm3x3 — COMMENTED** [review]
Just a few comments, haven't made it very far 😅

File: app/components/navigation/navigation-item.tsx (hunk 1 above)
Comment (internette) [reply app/components/navigation/navigation-item.tsx:6]: > nit: I would say formatting this so that it can be more friendly to read on narrow screens can be a nice optimization in code quality and review. It can also make diffs easier to read.

Agreed. I'll make this change here. I've also created an issue to tackle stuff like this with auto-formatting: https://github.com/jjm3x3/AmazingRaceFantasy/issues/97

File: app/components/navigation/navigation-item.tsx (hunk 3 above)
Comment (internette) [reply app/components/navigation/navigation-item.tsx:10]: Haha fair. I'm used to this being automatically added, so I'm still adjusting to manually adding it.

File: jest.config.mjs [hunk 4]
```diff
@@ -11,6 +11,8 @@ const config = {
   // setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
 
   testEnvironment: "jest-environment-jsdom",
+  setupFilesAfterEnv: ["<rootDir>/__tests__/setupTests.js"],
+  testPathIgnorePatterns: ["<rootDir>/build/", "<rootDir>/node_modules/", "<rootDir>/__tests__/setupTests.js"]
```
Comment (jjm3x3) [inline jest.config.mjs:15]: Interested to know how you arrived at these values. the `setupTests.js` seems fairly obvious, but the other ones are a bit interesting. Are those defaults that now need to be specified since we are including this config, or are those ones that are usually added and just where not added originally because I set it up and still learning 😅

File: app/page.tsx [hunk 5]
```diff
@@ -1,49 +1,33 @@
...
-    const currentDirFilesList = fs.readdirSync(__dirname)
-    let archiveDirFilesList: string[] = []
-
-    if (currentDirFilesList.includes("archive")) {
-        console.log("Yay there is an 'archive' folder and it's contents are:")
-        archiveDirFilesList = fs.readdirSync(__dirname+"/archive")
-        console.log(archiveDirFilesList)
-    }
-
+    const pages = getPages() || [];
```
Comment (jjm3x3) [inline app/page.tsx:8]: fft: this seems like a simple enough refactor that could have been pulled into a separate PR and not get wrapped up in all the bigger more important changes you are doing. 😅

**Review by jjm3x3 — COMMENTED** [review]
Just another round of comments, nearing half way through.

File: app/components/navigation/navigation-item.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+            type="checkbox"
+            onChange={(e)=> setIsHidden(!e.target.value)}
+        />
+        <label 
+            htmlFor={inputAttr.id}
+            id={labelAttr.id}
+            aria-controls={labelAttr.aria.controls}
+            data-testid={labelAttr.testId}
+            className={labelAttr.classes }>{labelAttr.content}</label>
+        {hasSubpages && <ul className={listAttr.classes}  id={listAttr.id} data-testid={listAttr.testId} aria-hidden={isHidden} hidden={isHidden}>{children}</ul>}
```
[the remark this answers is not in the document]
Comment (internette) [reply app/components/navigation/navigation-item.tsx:18]: This will be fixed in the auto-format ticket :)

File: app/components/navigation/navigation.module.scss
Comment (jjm3x3) [inline app/components/navigation/navigation.module.scss:1]: Not a blocker, but I have been trying to convert as many things to tailwind as I reasonably can. Might be worth checking out for consistency.

**Review by jjm3x3 — COMMENTED** [review]
Very close, just looking over the sass/css that we added for the nav...

File: app/components/navigation/navigation.module.scss [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        right: 0.5rem;
+    }
+}
+
+.sub-level-link {
+    padding: 0.25rem 0.5rem;
+    display: inline-block;
+    width: 100%;
+    &:hover {
+        cursor: pointer;
```
Comment (jjm3x3) [inline app/components/navigation/navigation.module.scss:37]: I actually think this is unnecessary, right?
```suggestion
```

**Review by jjm3x3 — APPROVED** [review]
I think this is looking good, just lots of questions and nit picks, but this does what we need it to do and quite well, Nice Job!

File: app/models/INavigationItem.tsx
[the remark this answers is not in the document]
Comment (internette) [reply app/models/INavigationItem.tsx:1]: I'll leave it as is right now. If we decide to change, we can refactor then

File: app/components/navigation/navigation.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+                id: `subpage-${keyName}-dropdown`,
+                testId: `subpage-${keyName}-dropdown`,
+                classes: styles['nav-subpages-list']
+            }
+            return (<li key={`nav-toplevellink-${keyName}`} className={styles['nav-toplevel-page']}>
+                <NavigationItem hasSubpages={hasSubpages }  inputAttr={subpageInputAttr } labelAttr={subpageLabelAttr } listAttr={ subpageListAttr} children={page.subpages.map((subpage: ISubpage) => {
+                        const subpageKeyName = subpage.name.toLowerCase().replaceAll(' ', '-');
+                        return <li key={`nav-toplevellink-${keyName}-sublink-${subpageKeyName}`}>
+                            <Link href={subpage.path} className={styles['sub-level-link']}>{subpage.name}</Link>
+                        </li>
```
[the remark this answers is not in the document]
Comment (internette) [reply app/components/navigation/navigation.tsx:56]: I'll create a separate PR for this
